fl4health/clients/fenda_client.py

```diff
@@ -46,13 +49,16 @@ def __init__(
             Each value associate with one of two contrastive losses in perfcl loss.
             cos_sim_loss_weight: Weight to be used for cosine similarity loss.
             contrastive_loss_weight: Weight to be used for contrastive loss.
+            mkmmd_loss_weight: Weight to be used for mkmmd loss.
         """
         self.perfcl_loss_weights = perfcl_loss_weights
         self.cos_sim_loss_weight = cos_sim_loss_weight
         self.contrastive_loss_weight = contrastive_loss_weight
-        self.cos_sim = torch.nn.CosineSimilarity(dim=-1).to(self.device)
-        self.ce_criterion = torch.nn.CrossEntropyLoss().to(self.device)
+        self.mkmmd_loss_weight = mkmmd_loss_weight
         self.temperature = temperature
+        self.cos_sim = torch.nn.CosineSimilarity(dim=-1).to(self.device)
+        self.contrastive_loss = ContrastiveLoss(self.device, temperature=self.temperature).to(self.device)
+        self.mkmmd_loss = MkMmdLoss(device=self.device, minimize_type_two_error=False).to(self.device)
```

Comment:
It's probably worth experimenting with this a bit, but I sort of imagined we'd have at least two MkMMD losses to more closely mimic the PerFCL setup. That is, on MkMMD loss to ensure the new global features don't move too far from the old global features (we would minimize the type two error there) and one to make sure the new global features and the new local features were far from each other (as you're doing here). We could theoretically add a third that tried to make sure the the new local features don't drift too far from the old local features.

Reply:
I agree I am just experimenting various option for fenda and moon